test/terra/backends/aer_simulator/test_circuit.py

```diff
@@ -0,0 +1,150 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2018, 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+AerSimulator Integration Tests
+"""
+from math import sqrt
+from ddt import ddt
+from qiskit import ClassicalRegister, QuantumCircuit, QuantumRegister
+from qiskit.circuit import CircuitInstruction
+from qiskit.circuit.library import Measure
+from test.terra.reference import ref_algorithms
+
+from test.terra.backends.simulator_test_case import (
+    SimulatorTestCase, supported_methods)
+
+
+@ddt
+class TestVariousCircuit(SimulatorTestCase):
+    """AerSimulator tests to simulate various types of circuits"""
+
+    @supported_methods(
+        ['automatic', 'statevector', 'density_matrix',
+         'matrix_product_state', 'extended_stabilizer', 'tensor_network'])
+    def test_quantum_register_circuit(self, method, device):
+        """Test circuits with quantum registers."""
+        
+        qubits = QuantumRegister(3)
+        clbits = ClassicalRegister(3)
+
+        circuit = QuantumCircuit(qubits, clbits)
+        circuit.h(qubits[0])
+        circuit.cx(qubits[0], qubits[1])
+        circuit.cx(qubits[0], qubits[2])
+
+        for q, c in zip(qubits, clbits):
+            circuit.append(CircuitInstruction(Measure(), [q], [c]))
```

Comment:
Is there a reason to do this instead of just?:

```suggestion
        for q, c in zip(qubits, clbits):
            circuit.measure(q, c)
```

Reply:
Because other codes in this test file call the same way :-) I will update whole measurement in this test file.